modules/osdk-monitoring-prometheus-servicemonitor-creating.adoc

```diff
@@ -0,0 +1,31 @@
+// Module included in the following assemblies:
+//
+// * applications/operator_sdk/osdk-monitoring-prometheus.adoc
+
+[id='osdk-monitoring-prometheus-servicemonitor-creating-{context}']
+= Creating ServiceMonitor resources
+
+Operator authors can add Service target discovery of created monitoring Services
+using the `metrics.CreateServiceMonitor()` helper function, which accepts the
+newly created Service.
+
+.Prerequisites
+
+- Go-based Operator generated using the Operator SDK
+- Kubernetes-based cluster with the Prometheus Operator deployed
+
+.Procedure
+
+. Add the `metrics.CreateServiceMonitor()` helper function, creating only one
+`ServiceMonitor` per application and namespace:
++
+[source,go]
+----
+    import "github.com/operator-framework/operator-sdk/pkg/metrics"
+
+    // Pass the Service(s) to the helper function, which in turn returns the `ServiceMonitor` object.
+    serviceMonitors, err := metrics.CreateServiceMonitors(restConfig, "namespace", services)
```

Comment:
@LiliC Should `"namespace"` be a replaceable? `services`?

Reply:
`services` are meant in this case an array of  kubernetes`Service` objects the user creates for metrics exposing.

Yes `"namespace"` is a silly example I guess, should include the signature of the function as well. It's the name of the `Namespace` in which the ServiceMonitor will be created.
```
CreateServiceMonitors(config *rest.Config, ns string, services []*v1.Service)
```